File: web/src/data/download.ts
import https from 'https';
import fs from 'fs';

const fileUrl = 'https://www.bls.gov/emp/skills/skills.xlsx';
const destination = `data/skills.xlsx`;

const file = fs.createWriteStream(destination);

https
  .get(fileUrl, (response) => {
    response.pipe(file);
    file.on('finish', () =>
      file.close(() => console.log('File downloaded successfully'))
    );
  })
  .on('error', (err) =>
    fs.unlink(destination, () => console.error('Error downloading file:', err))
  );

File: src/data/download.py
import requests


def download_data(url: str, xlsx_file: str):
    response = requests.get(url)
    with open(xlsx_file, "wb") as file:
        file.write(response.content)


if __name__ == "__main__":
    download_data(
        "https://www.bls.gov/emp/skills/skills.xlsx",
        "data/skills.xlsx")
